types/parameterized_list_type.go

```diff
@@ -0,0 +1,59 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/substrait-io/substrait-go/types/parameter_types"
+)
+
+// ParameterizedListType is a list type having parameter of ParameterizedAbstractType
+// basically a list of which type is another abstract parameter
+// example: List<Decimal(P,S)>. Kindly note concrete types List<Decimal(38, 0)> is not represented by this type
+// Concrete type is represented by ListType
+type ParameterizedListType struct {
+	Nullability      Nullability
+	TypeVariationRef uint32
+	Type             ParameterizedAbstractType
+}
+
+func (*ParameterizedListType) isRootRef() {}
+func (m *ParameterizedListType) WithNullability(n Nullability) Type {
+	m.Nullability = n
+	return m
+}
```

Comment:
value receiver

Reply:
I deliberately converted it to pointer receiver. This is to catch error of using wrong type at compile time instead of run time.
Basically, if a interface is implemented with a pointer receiver. Value receiver won't implement this.
Since pointer of ParameterizedListType implements "FuncDefArgType" so value receiver won't satisfy interface "FuncDefArgType"
I feel this is helpful if we have "FuncDefArgType" and need to switch case on "type". We can get compile error than
(sample program is [here](https://go.dev/play/p/-N-XAX8DrVm))

Let me know if this sounds ok